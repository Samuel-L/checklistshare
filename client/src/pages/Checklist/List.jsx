import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from 'material-ui/styles';
import Grid from 'material-ui/Grid';
import Typography from 'material-ui/Typography';
import Checkbox from 'material-ui/Checkbox';
import List, { ListItem, ListItemText } from 'material-ui/List';

const styles = theme => ({
  listGridRoot: {
    width: '100%',
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper,
  },
  checkbox: {
    color: '#57c557 !important',
  },
});

const ListComponent = props => (
  <Grid container justify="center">
    <Grid item xs={12}>
      <Typography
        variant="headline"
        align="center"
        gutterBottom
      >
        {props.checklist.title}
      </Typography>
    </Grid>
    <Grid item className={props.classes.listGridRoot}>
      <List>
        {
          props.checklist.items.map(item => (
            <ListItem
              key={item.id}
              dense
              button
              onClick={props.toggleList(item.id)}
            >
              <Checkbox
                className={props.classes.checkbox}
                checked={props.checked.indexOf(item.id) !== -1}
                tabIndex={-1}
                disableRipple
              />
              <ListItemText primary={item.name} />
            </ListItem>
          ))
        }
      </List>
    </Grid>
  </Grid>
);

ListComponent.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  checklist: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    url: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
      List: PropTypes.number,
      name: PropTypes.string,
    })),
  }).isRequired,
  toggleList: PropTypes.func.isRequired,
  checked: PropTypes.arrayOf(PropTypes.number).isRequired,
};

export default withStyles(styles)(ListComponent);
