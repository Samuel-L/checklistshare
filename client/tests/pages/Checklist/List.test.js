import React from 'react';
import renderer from 'react-test-renderer';

import List from '../../../src/pages/Checklist/List';

describe('pages/Checklist: List', () => {
  const classes = { listGridRoot: '', checkbox: '' };
  const checklist = {
    id: 0,
    title: '',
    url: '',
    items: [
      { id: 0, List: 0, name: '' },
    ],
  };
  const toggleList = jest.fn();
  const checked = [0];

  const tree = renderer.create(
    <List
      classes={classes}
      checklist={checklist}
      toggleList={toggleList}
      checked={checked}
      toggleEditMode={jest.fn()}
    />
  ).toJSON();

  it('matches snapshot', () => {
    expect(tree).toMatchSnapshot(); 
  });
});
