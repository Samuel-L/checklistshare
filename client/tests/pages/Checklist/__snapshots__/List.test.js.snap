// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Checklist: List matches snapshot 1`] = `
<div
  className="MuiGrid-container-3"
>
  <div
    className="MuiGrid-item-4 MuiGrid-grid-xs-12-40"
  >
    <h1
      className="MuiTypography-root-93 MuiTypography-headline-98 MuiTypography-gutterBottom-110 MuiTypography-alignCenter-106"
    >
      
    </h1>
  </div>
  <div
    className="MuiGrid-item-4 ListComponent-listGridRoot-1"
  >
    <ul
      className="MuiList-root-117 MuiList-padding-118"
    >
      <div
        className="MuiButtonBase-root-131 MuiListItem-root-121 MuiListItem-dense-125 MuiListItem-gutters-128 MuiListItem-button-129"
        onBlur={[Function]}
        onClick={undefined}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButtonBase-root-131 MuiIconButton-root-143 MuiSwitchBase-root-139 MuiCheckbox-root-134 MuiCheckbox-colorSecondary-138 MuiSwitchBase-checked-140 MuiCheckbox-checked-135 ListComponent-checkbox-2"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={undefined}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-148"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-149"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <input
              checked={true}
              className="MuiSwitchBase-input-142"
              disabled={undefined}
              id={undefined}
              name={undefined}
              onChange={[Function]}
              tabIndex={-1}
              type="checkbox"
              value={undefined}
            />
          </span>
        </span>
        <div
          className="MuiListItemText-root-155 MuiListItemText-dense-157"
        />
        <span
          className="MuiTouchRipple-root-161"
        />
      </div>
    </ul>
  </div>
</div>
`;
